src/components/NcContent/NcContent.vue

```diff
@@ -77,21 +88,48 @@ It includes the Navigation, the App content and the Sidebar.
 </template>
 
 <script>
+import { computed } from 'vue'
+
 export default {
+	provide() {
+		return {
+			isMobile: computed(() => this.isMobile),
+			isFullscreen: computed(() => this.isFullscreen),
+		}
+	},
 	props: {
 		appName: {
 			type: String,
 			required: true,
 		},
 	},
+	data() {
+		return {
+			isMobile: false,
+			isFullscreen: false,
+		}
+	},
+	created() {
+		window.addEventListener('resize', this.handleWindowResize)
+		this.handleWindowResize()
+	},
+	beforeUnmount() {
+		window.removeEventListener('resize', this.handleWindowResize)
+	},
+	methods: {
+		handleWindowResize() {
+			this.isMobile = document.documentElement.clientWidth < 1024
+			// if the window height is equal to the screen height,
+			// we're in full screen mode
+			this.isFullscreen = (window.outerHeight === screen.height)
```

Comment:
After dropping IE11 support, maybe we can use a native event?
https://caniuse.com/?search=fullscreenchange

Reply:
Fine with me, but for Safari on iOS we would seem to need a non-standard fallback.